### ISSUE TITLE
Strange values at symmetric BC when AXISYMMETRIC=YES

### ISSUE THREAD

**Issue by mskim0401** [su2code/SU2#1373]
Dear all,

I simulated Sym_plane_testcases from #657 
When I simulate axisymmetric probem (AXISYMMETRIC =YES), I found odd value at symmetric axis. This problem doesn't appear when I turn off AXISYMMETRIC option.

![Cp_comparison](https://user-images.githubusercontent.com/74228020/133757118-b17e8772-9919-4cb1-b1f2-8cf18e4a48ac.png)
========================================================================
========================================================================
.
.

![Density_comparison](https://user-images.githubusercontent.com/74228020/133757136-63ae13a2-7c3d-478f-9a81-7aaf001c6dd2.png)
========================================================================
========================================================================
.
.

![Mach_comparison](https://user-images.githubusercontent.com/74228020/133757154-9ed10855-3070-4210-b5b5-e2e3f2fad273.png)
========================================================================
========================================================================
.
.

![Temp_comparison](https://user-images.githubusercontent.com/74228020/133757157-18646978-20e6-49e7-a44f-ac45151b3901.png)

It seems that temperature, Mach, and momentum have no problem, but density, pressure, and energy are strange at the symmetric axis...
From #657 , I think it is flux related problem.

I attached version 7 configure file I used.
[Sym_BC_test.zip](https://github.com/su2code/SU2/files/7184390/Sym_BC_test.zip)

I'm also trying to simulate my axisymmetric problem and this problem appears too in my case.
![Base_Axis](https://user-images.githubusercontent.com/74228020/133761583-b3ea8fb4-876a-422c-a957-4a0ce1cc9783.png)
Symmetric BC is x-axis.

How I fix this problem? Could anyone help me?

Thank you.

**Desktop (please complete the following information):**
 - OS: Ubuntu 18.04
 - SU2 Version: v7.1.0

**Comment by pcarruscag** [comment]
Some issues were fixes is version 7.2.0 related with axisymmetry. Give it a try please.

**Comment by pcarruscag** [comment]
Correction, they were fixed in the develop branch, not released yet #1366

**Comment by mskim0401** [comment]
I tried to fix this problem about 10 days. I conducted code in #1366 but there still are problem at near axis. (In my experience, this bug affect not that big effect on solution)

One thing I found is that it can be improved by using Upwind scheme (e.g Roe, HLLC) but not exactly resolved.

I attached density profile using Roe scheme.
<Roe scheme Density Results>
![Density](https://user-images.githubusercontent.com/74228020/135422230-5193ca99-a58a-454a-b4d3-b62a6da5b44b.png)

**Comment by bigfooted** [comment]
I just ran the case and I see this as well, even after complete convergence. You clearly can see this in the energy and pressure (and density) but not in the momentum terms. The temperature field also looks smooth. So may be related to density/pressure.

**Comment by mskim0401** [comment]
What I found is that density unrelated values look smooth for example, velocities (I got it Momentum/Density) and total energy (Energy/Density).

As I understand, pressure is calculated assuming perfect gas:
![image](https://user-images.githubusercontent.com/74228020/135700745-843827fc-3b3c-4e4e-8d25-fd501d39f238.png)
and temperature is 
![image](https://user-images.githubusercontent.com/74228020/135700773-2657869c-6203-4107-a4b4-186e6cc697a1.png)
Pressure is function of density but temperature is not. I guess that's why temperature looks good.

However, I don't know why momentum has no problem.

**Comment by WallyMaier** [comment]
I wonder if this issue arises only with  AXISYMMETRIC = YES and a BC_SYM?

**Comment by mskim0401** [comment]
I just tested some boundary conditions on O cylinder case.
And I found this issue occurs at near axis when AXISYMMETRIC = YES regardless of which BC is applied at axis (y=0). For example, this also arises on BC_HeatFlux_Wall with zero heat flux.

**Comment by mskim0401** [comment]
Yes I tried both, however this issue still arose...

**Comment by mskim0401** [comment]
At least for my case, this problem doesn't have a significant impact on the entire solution.
In addition, this oscillating pattern appears about only 4 cells near axis regardless of how far it is from the axis.

**Comment by jtneedels** [comment]
@mskim0401 Yes, that is consistent with what we are seeing. Although the effect seems localized, it is disrupting the solution quality (particularly surface pressure and density) in the vicinity of the stagnation point. @WallyMaier and I are taking a look to isolate the cause.

**Comment by jtneedels** [comment]
@pcarruscag @mskim0401 @WallyMaier @bigfooted I see these oscillations persistently in non-axisym cases with MUSCL turned off. I went through the BC_SYM_PLANE implementation in CFVMFlowSolverBase.inl, but I couldn't find anything that affects this issue. It's interesting we've observed it in a variety of solvers, could it be how the node at the corner of the symm plane and wall is treated?

**Comment by ChristianBauerEng** [comment]
Not sure if this issue is related, but I'm performing axisymmetric simulations of a convergent nozzle. Throughout most of the domain the axis values seem fine, but there is a region where density behaves quite strangely at the axis.

What is seen here is the density gradient, just to make it more visible:
![grafik](https://user-images.githubusercontent.com/42602495/152206391-755443cc-7c7e-43f1-812a-2335879027c4.png)

Results obtained with SLAU2, 2nd order euler.

**Comment by pcarruscag** [comment]
Trying to leave something other than the usual "go away stale bot", one theory from the last dev meeting is that the gradients used in the axisymmetric source may need some treatment consistent with the symmetry boundary.

**Comment by antares190** [comment]
Just my 2¢ here. I've been struggling with a relatively simple simulation, Euler, 2D axisymmetric, supersonic, AUSM (SU2 7.5.1). I tried several meshes (and meshers) and always diverged no matter what (quality is ok, CFL as well). Conditions were ok, and were mimicking  inv_wedge tutorial (which, btw was running ok even with axisymmetry ON).

Long story short, I opened with a text editor the original geometry STEP file, and noticed it was carrying from the CAD some (engineering wise negligible) numerical terms (say, point 0, 0 was actually 0, 1e-6). Hence the symmetry axis was somewhat off. By correcting the STEP file, everything went fine.

I wonder whether this is the expected behavior of this kind of simulation, and / or if there is any artificial diffusion parameter (such as ENTROPY_FIX_COEFF) that actually can sort things out for the EULER mode.

Did any of you experienced anything similar?

Thanks!

**Comment by pcarruscag** [comment]
Axisymmetry on a node-based code requires special treatment near the axis because some terms are 1/r and so if r is close to zero... If your geometry had the axis at 1e-6 the "axis treatment" was not being applied.

**Comment by antares190** [comment]
Thank you Pedro for the lightning reply! 
Yes, I see, unfortunately, I can't cross-check the log right now to see whether the "STRAIGHT EDGE" string was triggered, but I'll pay attention to that in the future for sure. 

Anyway, I wonder whether the axis treatment is looking for a "numerical 0", or does perform some sort of filtering or tolerance on the mesh ( in some ways similar to what some CADs do).

In my case, the body was in the order of meters, and the domain was in the order of 100 m.

**Comment by bigfooted** [comment]
ANSYS Fluent for instance has the same strict criterion, the axisymmetric line should be placed at the x-axis with high precision. In SU2, 
y < 1.0e-16 for the y-coordinates of the nodes on the symmetry axis

**Comment by bigfooted** [comment]
This bug was supposed to have been solved by #2194, but a recent test shows that this issue still exists.
